cmd/compact.go

```diff
@@ -47,7 +47,7 @@ func NewCompactCommand(ctx context.Context) *cobra.Command {
 			}
 
 			cp := compactor.NewCompactor(store, logrus.NewEntry(logger))
-			snapshot, err := cp.Compact(options, opts.needDefragmentation)
+			snapshot, err := cp.Compact(options, opts.needDefragmentation, opts.snapshotTimeout.Duration, opts.defragTimeout.Duration)
```

Comment:
I am thinking if we could aggregate `needDefragmentation` and `defragTimeout` to a structure and pass the structure to Compact function. WDUT @ishan16696 ?

Reply:
I remember having this discussion during the first PR for `compact` sub-command. How about introducing a `CompactOptions` struct which embeds `RestoreOptions` and add the additional flags?

Reply:
I prefer to aggregate all these flags `needDefragmentation`,`snapshotTimeout` and `defragTimeout` into a struct and kept `RestoreOptions` as separate in `CompactOptions` as [here](https://github.com/gardener/etcd-backup-restore/blob/master/cmd/options.go#L130-L133).
WDYT?

Reply:
> I prefer to aggregate all these flags needDefragmentation,snapshotTimeout and defragTimeout into a struct and kept RestoreOptions as separate in CompactOptions as here.

Makes sense. But you might need a different struct in (or move the struct you pointed to to) the `types` package.

Reply:
Yes

Reply:
I have added the necessary changes in the compaction code. Please PTAL !!